portal-impl/src/com/liferay/portal/servlet/ComboServlet.java

```diff
@@ -307,7 +307,7 @@ protected void doService(
 			if (cacheEnabled && (modulePathsString != null) &&
 				!PropsValues.COMBO_CHECK_TIMESTAMP) {
 
-				if (PropsValues.COMBO_MAX_FILES != -1) {
+				if (modulePaths.length <= PropsValues.COMBO_MAX_FILES) {
```

Comment:
This check looks ok to me, and it makes unnecessary the computation of `totalFilesCount` and the subsequent second comparison with `PropsValues.COMBO_MAX_FILES`